modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -172,6 +174,27 @@ public void onFailure(Exception e) {
                                     return;
                                 }
                             }
+                            String body;
+                            try {
+                                body = EntityUtils.toString(re.getResponse().getEntity());
+                            } catch (IOException ioe) {
+                                ioe.addSuppressed(e);
+                                fail.accept(ioe);
```

Comment:
not terribly sure about this. we could still recover the right status code although there was a problem reading the body... not sure which exception should win :)